reactive-streams/src/main/scala/fs2/interop/reactivestreams/StreamSubscriber.scala

```diff
@@ -53,7 +55,14 @@ final class StreamSubscriber[F[_]: ConcurrentEffect, A](val sub: StreamSubscribe
 
 object StreamSubscriber {
   def apply[F[_]: ConcurrentEffect, A]: F[StreamSubscriber[F, A]] =
-    fsm[F, A].map(new StreamSubscriber(_))
+    Queue
+      .bounded[F, Either[Throwable, Option[A]]](2)
+      .flatMap(fsm[F, A](_, 1).map(new StreamSubscriber(_)))
+
+  def apply[F[_]: ConcurrentEffect, A](maxDemand: Int): F[StreamSubscriber[F, A]] =
+    Queue
+      .bounded[F, Either[Throwable, Option[A]]](maxDemand * 2)
```

Comment:
Why is the queue sized with twice the demand?

Reply:
We could theoretically have `maxDemand + 1 (an exception) + 1 (None termination)` elements in the queue. So we need a `maxDemand + 2` sized queue. I updated the queue size to `maxDemand + 2`